### ISSUE TITLE
Latest nightly is not working

### ISSUE THREAD

**Issue by CountParadox** [Tzahi12345/YoutubeDL-Material#146]
Hey @Tzahi12345 , 

I just pulled the latest nightly, and my logs are filled with 

```

chown: ./subscriptions/playlists/DL-List/Bulletproof [AMV].mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/[Electro Swing Remix] Spooky Scary (Swingin') Skeletons - Extended Club Version.info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/【AMV】Soul Eater『This Is Halloween』.jpg: Operation not permitted

```

and im unable to load the web UI...

**Comment by CountParadox** [comment]
another chunk

```

chown: ./subscriptions/playlists/DL-List/AMV Cowboy Bebop  Feat C2C Down the Road.jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/j^p^n - bloom. [].jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Jojo's Bizarre Adventure- Awaken(Pillar Men Theme).jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Samurai Champloo - Shiki No Uta (HQ).jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Jojo's Bizarre Adventure- Awaken(Pillar Men Theme).mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Light It UP!.mp4.f140: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Samurai Champloo - Shiki No Uta (HQ).info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Samurai Champloo - Shiki No Uta (HQ).mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Dance Monkey - AMV - 「Anime MV」.info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Dance Monkey - AMV - 「Anime MV」.jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Dance Monkey - AMV - 「Anime MV」.mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/j^p^n - bloom. [].info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/j^p^n - bloom. [].mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Dance Monkey - AMV - 「Anime MV」.mp4.f140: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/AMV Cowboy Bebop  Feat C2C Down the Road.info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/j^p^n - bloom. [].mp4.f140: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/AMV Cowboy Bebop  Feat C2C Down the Road.mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Evangelion AMV _ $UICIDEBOY$- Antartica, Ugly, Jeffer Drive.jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Evangelion AMV _ $UICIDEBOY$- Antartica, Ugly, Jeffer Drive.info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/.nfs000000000000922b0000002c: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Evangelion AMV _ $UICIDEBOY$- Antartica, Ugly, Jeffer Drive.mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Evangelion AMV _ $UICIDEBOY$- Antartica, Ugly, Jeffer Drive.mp4.f140: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Zamir & Marc Indigo – On The Move (feat. Chevy) (Lyrics) [CC].info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Zamir & Marc Indigo – On The Move (feat. Chevy) (Lyrics) [CC].jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Zamir & Marc Indigo – On The Move (feat. Chevy) (Lyrics) [CC].mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Bulletproof [AMV].info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/I Built an Unethical Zoo Where Nobody Is Safe - Planet Zoo.mp4.f140: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Bulletproof [AMV].mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Bulletproof [AMV].jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Zamir & Marc Indigo – On The Move (feat. Chevy) (Lyrics) [CC].mp4.f140: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Cab Calloway - St. James Infirmary (Remix) Feat. Harald Kindseth [ Clear Bass Boosted ].info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Cab Calloway - St. James Infirmary (Remix) Feat. Harald Kindseth [ Clear Bass Boosted ].mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Cab Calloway - St. James Infirmary (Remix) Feat. Harald Kindseth [ Clear Bass Boosted ].jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/[MV] HyunA(현아) _ Lip & Hip.info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/[MV] HyunA(현아) _ Lip & Hip.jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Light It UP!.info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Light It UP!.jpg: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Light It UP!.mp4: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/Jojo's Bizarre Adventure- Awaken(Pillar Men Theme).info.json: Operation not permitted,
chown: ./subscriptions/playlists/DL-List/[MV] HyunA(현아) _ Lip & Hip.mp4: Operation not permitted

```

**Comment by Tzahi12345** [comment]
Weird, I'm not reproducing. Maybe something went wrong with your image? Trying [deleting your nightly image](https://docs.docker.com/engine/reference/commandline/image_rm/) and re-pulling. Let me know if that helps!

**Comment by SuperSandro2000** [comment]
You either changed the user to a user which is not permitted to chown or dropped the CAP for chown.

**Comment by CountParadox** [comment]
> You either changed the user to a user which is not permitted to chown or dropped the CAP for chown.

Ive done neither, I just pulled the latest image and restarted the container...



> Weird, I'm not reproducing. Maybe something went wrong with your image? Trying [deleting your nightly image](https://docs.docker.com/engine/reference/commandline/image_rm/) and re-pulling. Let me know if that helps!

strange... I have deleted all the downloaded images, repulled the latest nightly and it continues like this...

**Comment by CountParadox** [comment]
OK, 
I added my UID and GID to the FSTAB mount, it seems to be a problem stemming from the fact that I have passed through a windows network share to the container... it wasnt an issue before though?

I also changed the port mapping INSIDE the container from 443 to 14472 and I can now access the web interface. It appears to have lost the configuration / users hmm

**Comment by SuperSandro2000** [comment]
I can't help you with docker on windows.

**Comment by CountParadox** [comment]
> docker on windows.

It's not on windows, it's on Ubuntu. The data is on a network share mounted over samba from a Synology NAS

**Comment by Tzahi12345** [comment]
@CountParadox Can you try setting the UID and GID to root? At least as a hotfix until I can reproduce and start seeing what's going on.

**Comment by CountParadox** [comment]
Oh it doesn't like that! I set them both to 0 and now there's NO logs, it's
as if it's not starting..?

**Comment by CountParadox** [comment]
To replicate, setup a samba share on your network, and mount it using
fstab.

Then start the container with binds to the network share mount

**Comment by Tzahi12345** [comment]
I'm unfortunately not able to set one up right now (maybe in the next few days I'll get a chance). I guess the root method didn't work, how about changing [how you mount?](https://devops.stackexchange.com/a/4439). A bit strange it didn't like the root user, maybe it was running a long chown? It had to be doing something.

**Comment by fluxtendu** [comment]
I have the same issue with a Docker container and a NFS share. 
To fix it I have commented this part of entrypoint.sh:
```
if [ "$*" = "$CMD" ] && [ "$(id -u)" = "0" ]; then
  find . \! -user "$UID" -exec chown "$UID:$GID" -R '{}' +
  exec su-exec "$UID:$GID" "$0" "$@"
fi
```
I don't exactly understand what this code do, but I don't think it should be mandatory to own the directory where the files are downloaded (and all the files in it). 

Maybe a write access test could be useful, but even this shouldn't stop the container to run...

**Comment by SuperSandro2000** [comment]
>  don't exactly understand what this code do

If you are executing the default permission and the script is run as root the permissions are changed to the su-exec user we drop to later.

> mandatory to own the directory

If you change the user it is and many docker library images do it, too.

> Maybe a write access test could be useful

This is probably not enough and we could just add ``|| true`` to bypass the chown if it fails.

**Comment by fluxtendu** [comment]
I'll be glad with a `|| true` bypass (but therefore it would no longer be mandatory...)

But beyond that I really don't understand security or usability issue to have a download folder owned by another user. (And even, ideally, we should be able to choose the destination on demand...)

It's another story for settings folders like appdata or subscriptions.

**Comment by Tzahi12345** [comment]
Fixed! Check out #185. Combined @fluxtendu's solution with a suggestion from @slaecker for replacing `|| true` with an echo. Sorry this took so long to fix, hope it works well now!

**Comment by CountParadox** [comment]
works a treat! im up and running again :)))
